### DISCUSSION TITLE
Block Protocol support?

### DISCUSSION THREAD

**Discussion by skyfaller** [CloudCannon/bookshop#90] [Ideas]
Hi folks, I was just wondering whether you've heard of the Block Protocol project to standardize open source web editor blocks, and whether you'll consider supporting it, especially once it is more mature: https://www.joelonsoftware.com/2022/01/27/making-the-web-better-with-blocks/

**Comment by bglw** [comment]
Hello!

I haven't seen this before, but it does look interesting. 

What would support ideally look like for you? It looks like the blocks are primarily React components, so I don't quite know how this would integrate with the many SSGs Bookshop supports.

Keen to hear your thoughts!

**Reply by skyfaller** [reply]
I'm just going to admit ignorance and not pretend to fully understand the Block Protocol yet, so take anything I say with a grain of salt. (I also don't understand React, for that matter.) For the record, Block Protocol is in early alpha, so if you think it's heading in the wrong direction you should give them feedback.

That said, from their documentation, here's [what's required of an app embedding blocks](https://blockprotocol.org/spec/embedding-applications):
1. importing the source code for the blocks it wants to display (whether at runtime or compiled along with its own source)
2. combining the source with the data properties the block requires (how exactly this works depends on the rendering context)
3. rendering the block on a webpage (or other application)
4. providing any external dependencies the block requires

It seems the desired end goal would be, if a user imported their website from another app supporting blocks (let's say Wordpress) into your website editor, they would still see the same exact blocks. So if they had a custom block for e.g. embedding a Bandcamp music player, they would still be able to edit that music player and move it around their website, and it would still publish working HTML/CSS/Javascript to their website. If in the future they discovered a cool block for e.g. animated SVGs on the Wordpress block hub, they'd be able to install it in your website editor even though it wasn't specifically written for your editor, because it's standardized.

**Reply by skyfaller** [reply]
It appears that [Block Protocol does *not* require React](https://hash.dev/blog/build-blocks-in-any-language).

> Most [#BlockProtocol](https://twitter.com/hashtag/BlockProtocol?src=hashtag_click) examples are in JavaScript/TypeScript, but you can build blocks in *any language* which supports JavaScript as a transpilation target 🤯 even [#fsharp](https://twitter.com/hashtag/fsharp?src=hashtag_click)!
-- https://twitter.com/BlockProtocol/status/1537468344810479618

> It's now a lot easier to create blocks that are not tied to JavaScript, TypeScript, or React. As long as a block can use browser Events, it can implement the Block Protocol.
